src/qml/imports/QFieldControls/+Qt6/CodeReader.qml

```diff
@@ -27,22 +29,63 @@ Popup {
   dim: true
 
   onAboutToShow: {
+    // when NFC is not accessible, make sure the only option, QR, is active
+    if (!withNfc && !settings.cameraActive) {
+      settings.cameraActive = true
+    }
     barcodeDecoder.clearDecodedString();
   }
 
   onAboutToHide: {
+    if (cameraLoader.active) {
+      cameraLoader.item.camera.torchMode = Camera.TorchOff
+    }
+  }
+
+  Settings {
+    id: settings
+    property bool cameraActive: true
+    property bool nearfieldActive: true
   }
 
   BarcodeDecoder {
     id: barcodeDecoder
 
     onDecodedStringChanged: {
       if (decodedString !== '') {
+        codeReader.decodedString = decodedString
         decodedFlashAnimation.start();
       }
     }
   }
 
+  Loader {
+    id: nearfieldLoader
+    active: withNfc && codeReader.openedOnce && settings.nearfieldActive
+
+    sourceComponent: Component {
+      Item {
+        id: nearFieldContainer
+
+        Component.onCompleted: {
+          Qt.createQmlObject('import org.qfield 1.0
```

Comment:
Do we have another option than plain text for this?

Reply:
@m-kuhn , we can't for for a plain text here as the NFC support is not hard-coded. For builds (i.e. linux, windows, macos) without NFC support, the NearFieldObject item isn't available and QML fails to load the CodeReader.qml, hence why we use a string.

An alternative here would be to create a dummy non-functional NearFieldObject for builds that are WITH_NFC=OFF.

Reply:
@m-kuhn , alternative implemented.